examples/text-generation/utils.py

```diff
@@ -557,6 +557,14 @@ def setup_tokenizer(args, model, assistant_model):
         tokenizer.eos_token = tokenizer.decode(tokenizer.eos_token_id)
         tokenizer.bos_token = tokenizer.decode(tokenizer.bos_token_id)
 
+    # HACK: MiniCPM3 has multiple eos_tokens and does not specify padding token. Set both to second one.
+    if model.config.model_type == "minicpm3":
+        tokenizer.pad_token = tokenizer.eos_token
+        model.generation_config.pad_token_id = model.generation_config.eos_token_id[-1]
+        model.generation_config.eos_token_id = model.generation_config.eos_token_id[-1]
+        if len(model.generation_config.eos_token_id) > 1:
+            print("WARNING: Multiple EOS token IDs found. Only last eos token id will be used.")
```

Comment:
```suggestion
            logger.warning("Multiple EOS token IDs found. Only last eos token id will be used.")
```
And you can pass the logger as an argument to the `setup_tokenizer` method as done here for instance: https://github.com/huggingface/optimum-habana/blob/9a492005f26b1be44f77b757914f40e4e39d033f/examples/text-generation/utils.py#L216

Reply:
Done